packages/core-mobile/app/store/notifications/listeners/manageForegroundNotificationSubscription.ts

```diff
@@ -0,0 +1,13 @@
+import { AppListenerEffectAPI } from 'store/index'
+import FCMService from 'services/fcm/FCMService'
+import { isAnyOf } from '@reduxjs/toolkit'
+import { onAppLocked, onBackground } from 'store/app'
+
+export async function manageForegroundNotificationSubscription(
+  listenerApi: AppListenerEffectAPI
+): Promise<void> {
+  const { condition } = listenerApi
+  const unsubscribe = FCMService.listenForMessagesForeground()
+  await condition(isAnyOf(onBackground, onAppLocked))
+  unsubscribe()
```

Comment:
is this logic necessary? what are we trying to prevent?

Reply:
listenForMessagesForeground registers fcm foreground listener that catches notifications sent from firebase. so when app goes to background we want to unsubscribe from it (there's separate listener for when you get notification while app in background, it's just how firebase sdk handles it)
But i think onAppLocked might be excess here. Will revisit that